ethereum/core/src/main/java/org/hyperledger/besu/ethereum/privacy/DefaultPrivacyController.java

```diff
@@ -284,16 +286,13 @@ public PrivacyGroup retrievePrivacyGroup(final String privacyGroupId, final Stri
       final RLPInput rlpInput =
           RLP.input(privateTransactionSimulatorResultOptional.get().getOutput());
       if (rlpInput.nextSize() > 0) {
-        return decodeList(rlpInput.raw());
+        return new PrivacyGroup(
+            privacyGroupId.toBase64String(), Type.ONCHAIN, "", "", decodeList(rlpInput.raw()));
       } else {
-        return Collections.emptyList();
+        return null;
```

Comment:
Can we change this method to return an Optional?

Reply:
done